presto-main/src/main/java/com/facebook/presto/execution/UseTask.java

```diff
@@ -52,12 +60,28 @@ public ListenableFuture<?> execute(
     {
         Session session = stateMachine.getSession();
 
+        TransactionId transactionId = session.getTransactionId().get();
+
+        Identity identity = session.getIdentity();
+
+        AccessControlContext context = session.getAccessControlContext();
+
         checkCatalogAndSessionPresent(statement, session);
 
         checkAndSetCatalog(statement, metadata, stateMachine, session);
 
         checkAndSetSchema(statement, metadata, stateMachine, session);
 
+        if (!hasCatalogAccess(identity, context, catalog, accessControl)) {
+            denyCatalogAccess(catalog);
+        }
+
+        CatalogSchemaName name = new CatalogSchemaName(catalog, schema);
```

Comment:
nit: since it's used only once, we can directly create this in the exception.

Reply:
refactored the code accordingly